lru.go

```diff
@@ -33,8 +33,6 @@ func (lru *lruCache[K, V]) add(newitem slruItem[K, V]) (oitem slruItem[K, V], ev
 	e := lru.ll.Back()
 	item := e.Value
 
-	delete(lru.data, item.key)
-
```

Comment:
Removed this because the same map entry gets assigned right below, there's no need to delete the key.

@dgryski to double-check that I'm not missing something here.